docs/notebooks/ask_tell_optimization.pct.py

```diff
@@ -0,0 +1,154 @@
+# %% [markdown]
+# # Ask-Tell Optimization Interface
+
+# %% [markdown]
+# In this notebook we will illustrate the use of Ask-Tell interface in Trieste. This is a useful interface to have in the cases when you want to have greater control of the optimization loop, or if letting Trieste manage this loop is impossible.
+#
+# First, some commmon code we will be using in the notebook.
+
+# %%
+import numpy as np
+import tensorflow as tf
+import matplotlib.pyplot as plt
+import gpflow
+
+from trieste.ask_tell_optimization import AskTellOptimizer
+from trieste.bayesian_optimizer import OptimizationResult, Record
+from trieste.data import Dataset
+from trieste.models.gpflow import GPflowModelConfig
+from trieste.objectives import scaled_branin, SCALED_BRANIN_MINIMUM
+from trieste.objectives.utils import mk_observer
+from trieste.observer import OBJECTIVE
+from trieste.space import Box
+
+from util.plotting import plot_regret
+
+np.random.seed(1234)
+tf.random.set_seed(1234)
+
+
+search_space = Box([0, 0], [1, 1])
+n_steps = 5
+
+def build_model(data, kernel_func=None):
+    """kernel_func should be a function that takes variance as a single input parameter"""
+    variance = tf.math.reduce_variance(data.observations)
+    if kernel_func is None:
+        kernel = gpflow.kernels.Matern52(variance=variance)
+    else:
+        kernel = kernel_func(variance)
+    gpr = gpflow.models.GPR(data.astuple(), kernel, noise_variance=1e-5)
+    gpflow.set_trainable(gpr.likelihood, False)
+
+    return GPflowModelConfig(**{
+        "model": gpr,
+        "optimizer": gpflow.optimizers.Scipy(),
+    })
+
+num_initial_points = 5
+initial_query_points = search_space.sample(num_initial_points)
+observer = mk_observer(scaled_branin)
+initial_data = observer(initial_query_points)
+model = build_model(initial_data)
+
+# %% [markdown]
+# ## Timing acquisition function: simple use case for Ask-Tell
```

Comment:
this is nice but less important than the other two use cases, I would put this one at the end

Reply:
the order here is from the simplest to more advanced, as is befitting the tutorial